### DISCUSSION TITLE
Why do I got this secondary temperature  by MCMC?

### DISCUSSION THREAD

**Discussion by Yzxastr** [phoebe-project/phoebe2#677] [Questions]
I try to get the median and uncertainty of secondary temperature by 'b.uncertainties_from_distribution_collection', but I got a temperature that's different from the normal, like this 
![secondary temperature](https://user-images.githubusercontent.com/102363557/208097859-66775b3b-b95a-4add-9fea-68a917557b5e.png)
From what I understand, the temperature of the secondary star is around 6,000 K. Could you tell me how to understand this secondary temperature?

**Comment by kecnry** [comment]
In order for this to not raise a passbands error, I'm guessing you're using blackbody atmospheres?  It is very difficult to tell what is happening without looking at the script and MCMC diagnostic plots, but my guess would be that the temperature is free to move across the entire range and is overcompensating for either a wrong starting position in another parameter or inflexibility in the other free parameters to account for some feature in the residuals.  In other words, the easiest way for MCMC to account for the data with the model and the free parameters you provided is by making your secondary star have essentially no light.  My suggestions would be to optimize the initial distributions such that you're fairly certain the true solution falls within them, inspect the residuals, and make sure that you're freeing up enough parameters to account for them.  You can also use priors to enforce that the temperature stays within certain "physical" bounds, but likely just doing that would result in all the samples "piling up" against whatever lower limit you set, which still won't give you reliable results.

**Comment by Yzxastr** [comment]
First of all, thank you for your suggestion. I will try to optimize the script according to your suggestions. I send my script and bundle to you and hope you can help me find the cause of the abnormal temperature of the secondary star. 
[my script and bundle.zip](https://github.com/phoebe-project/phoebe2/files/10260814/my.code.and.bundle.zip)

Then, I would like to ask you for advice on how to get a distribution image that is close to a circle ellipse like yours
![yours](https://user-images.githubusercontent.com/102363557/208469688-30be780c-a0ec-43e3-a4d3-fa0b228f9187.jpg)

 Now I run out like this.
![myMCMC](https://user-images.githubusercontent.com/102363557/208469785-c3a6d540-6fef-4e8a-a63f-3942d67c6463.png)
It seems to be a discrete distribution
Can you give me some advice on parameter setting?
Finally, thank you for your kind guidance.

**Reply by kecnry** [reply]
It really is hard to give any specific advice as each situation is different and my best advice would be to run through all the relevant tutorials and practice on similar fake data or systems with published solutions and make sure you can reproduce reasonable results and start to get a feel for what can easily go wrong.  You may also find some of the content from [our most recent workshop](http://phoebe-project.org/workshops/2022june/materials) to be useful, particularly "part 2" which covers fitting - but note that unlike the tutorials, these will not be kept up to date with future changes to phoebe.  

With all of that said, if this corner plot is from the same results as your secondary temperature, then with a temperature ratio of ~0.9, both the primary and secondary temperature are unphysically too cool, so that needs to be corrected first.  Your mass ratio is also quite small - is that expected for this system? You need starting distributions that are both physically reasonable and hopefully surround the true solution before you can expect to get good results back from MCMC.  If the walkers migrated to unphysical parameter space from reasonable starting points, then that either means you need to introduce other free parameters and/or apply reasonable priors.  If you're still seeing discrete/non-smooth distributions even after the values themselves make sense, you may just need more samples to fillout the distribution better, but if all the walkers are getting stuck (look at the [other emcee diagnostic plot "styles"](http://phoebe-project.org/docs/2.4/tutorials/emcee) to see what is happening to individual walkers), then that likely won't help.  Your resulting uncertainties also look very small, which could be a sign of underestimated uncertainties on the input data and could also be the cause of walkers getting stuck, if that is in fact what is happening (there are several tutorials and workshop tutorials that touch on this as well).

**Comment by Yzxastr** [comment]
OK,thank you so much.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "phoebe-project/phoebe2"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月21日(星期三) 晚上9:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [phoebe-project/phoebe2] Why do I got this secondary temperature  by MCMC? (Discussion #677)





 
It really is hard to give any specific advice as each situation is different and my best advice would be to run through all the relevant tutorials and practice on similar fake data or systems with published solutions and make sure you can reproduce reasonable results and start to get a feel for what can easily go wrong.  You may also find some of the content from our most recent workshop to be useful, particularly "part 2" which covers fitting - but note that unlike the tutorials, these will not be kept up to date with future changes to phoebe.
 
With all of that said, if this corner plot is from the same results as your secondary temperature, then with a temperature ratio of ~0.9, both the primary and secondary temperature are unphysically too cool, so that needs to be corrected first.  Your mass ratio is also quite small - is that expected for this system? You need starting distributions that are both physically reasonable and hopefully surround the true solution before you can expect to get good results back from MCMC.  If the walkers migrated to unphysical parameter space from reasonable starting points, then that either means you need to introduce other free parameters and/or apply reasonable priors.  If you're still seeing discrete/non-smooth distributions even after the values themselves make sense, you may just need more samples to fillout the distribution better, but if all the walkers are getting stuck (look at the other emcee diagnostic plot "styles" to see what is happening to individual walkers), then that likely won't help.  Your resulting uncertainties also look very small, which could be a sign of underestimated uncertainties on the input data and could also be the cause of walkers getting stuck, if that is in fact what is happening (there are several tutorials and workshop tutorials that touch on this as well).
